client/packages/system/src/Report/components/ReportSelector.tsx

```diff
@@ -57,42 +77,52 @@ export const ReportSelector: FC<PropsWithChildren<ReportSelectorProps>> = ({
   const hasPermission = !isLoading && data !== undefined;
   const noReports = !isLoading && !data?.nodes?.length;
   const oneReport =
-    !isLoading && data?.nodes?.length === 1 ? data.nodes[0] : null;
+    !isLoading && data?.nodes?.length === 1 ? data.nodes[0] : undefined;
 
-  return !!oneReport ? (
-    <div onClick={() => onClick(oneReport)}>{children}</div>
-  ) : (
-    <PaperClickPopover
-      placement="bottom"
-      width={350}
-      Content={
-        <PaperPopoverSection
-          label={t('select-report')}
-          labelStyle={{ width: '100%' }}
-          alignItems="center"
-        >
-          {isLoading ? (
-            <CircularProgress size={12} />
-          ) : (
-            <Box
-              style={{
-                maxHeight: '200px',
-                overflowY: 'auto',
-              }}
-              display="flex"
-              flexDirection="column"
+  return (
+    <>
+      {!!oneReport ? (
+        <div onClick={() => onReportSelected(oneReport)}>{children}</div>
+      ) : (
+        <PaperClickPopover
+          placement="bottom"
+          width={350}
+          Content={
+            <PaperPopoverSection
+              label={t('select-report')}
+              labelStyle={{ width: '100%' }}
+              alignItems="center"
             >
-              {noReports ? (
-                <NoReports hasPermission={hasPermission} />
+              {isLoading ? (
+                <CircularProgress size={12} />
               ) : (
-                reportButtons
+                <Box
+                  style={{
+                    maxHeight: '200px',
+                    overflowY: 'auto',
+                  }}
+                  display="flex"
+                  flexDirection="column"
+                >
+                  {noReports ? (
+                    <NoReports hasPermission={hasPermission} />
+                  ) : (
+                    reportButtons
+                  )}
+                </Box>
               )}
-            </Box>
-          )}
-        </PaperPopoverSection>
-      }
-    >
-      {children}
-    </PaperClickPopover>
+            </PaperPopoverSection>
+          }
+        >
+          {children}
+        </PaperClickPopover>
+      )}
+
+      <ReportArgumentsModal
+        report={reportWithArgs}
+        onReset={() => setReportWithArgs(undefined)}
+        onArgumentsSelected={(report, args) => onClick(report, args)}
```

Comment:
no need to create a new function - can just use `onArgumentsSelected={onClick}`

Reply:
done!